File: JavaArchitecture/app/src/main/java/com/demo/architecture/ui/main/MainActivity.java
package com.demo.architecture.ui.main;

import android.app.Activity;
import android.os.Bundle;

import com.alibaba.android.arouter.facade.Postcard;
import com.alibaba.android.arouter.facade.annotation.Autowired;
import com.alibaba.android.arouter.facade.annotation.Route;
import com.alibaba.android.arouter.facade.callback.NavigationCallback;
import com.alibaba.android.arouter.launcher.ARouter;
import com.demo.architecture.R;
import com.demo.architecture.base.BaseMvpActivity;
import com.demo.architecture.base.ComponentHolder;
import com.demo.architecture.base.Constants;
import com.demo.architecture.injection.component.DaggerMainActivityComponent;
import com.demo.architecture.injection.module.MainActivityMoudel;
import com.demo.architecture.model.UpdateModel;
import com.demo.architecture.utils.L;

/**
 * Created by HeYingXin on 2018/1/19.
 */
@Route(path = Constants.Route.MAIN_ACTIVITY, group = Constants.Route.GROUP_ONE)
public class MainActivity extends BaseMvpActivity<MainPersenter> implements MainContract.View{

    @Autowired()
    long key1 ;

    @Autowired()
    String key3 ;

    public static void startMainActivity() {
        ARouter.getInstance().build(Constants.Route.MAIN_ACTIVITY,Constants.Route.GROUP_ONE).withLong("key1", 666L)
                .withString("key3", "888").navigation(ComponentHolder.getAppComponent().context(), new NavigationCallback() {
            @Override
            public void onFound(Postcard postcard) {
                L.e("zhao", "onArrival: 找到了 ");
            }

            @Override
            public void onLost(Postcard postcard) {
                L.e("zhao", "onArrival: 找不到了 ");
            }

            @Override
            public void onArrival(Postcard postcard) {
                L.e("zhao", "onArrival: 跳转完了 ");
            }

            @Override
            public void onInterrupt(Postcard postcard) {
                L.e("zhao", "onArrival: 被拦截了 ");
            }
        });
    }

    public static void startMainActivity2(Activity activity) {
        //直接通过所有拦截器并且制定requestCode
        ARouter.getInstance().build(Constants.Route.MAIN_ACTIVITY,Constants.Route.GROUP_ONE).withLong("key1", 666L)
                .withString("key3", "888").greenChannel().navigation(activity,Constants.IntentRequest.MAIN_CODE);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setWindowlucency();
        setContentView(R.layout.activity_main);
        mPresenter.requestPermissions();
    }

    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    protected void onStop() {
        super.onStop();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    @Override
    protected void initView() {
        setEnableSwipe(false);
    }

    @Override
    protected void initData() {
        mPresenter.updateAPK();
    }

    @Override
    protected void getmPresenter() {
        //方式一，Inject无构造
//        activityComponent.inject(this);//注意这个方法里面需要打开一个inject,并且Presenter无法传参，传参也得传Inject的参数
        //方式二(构造加String)
        DaggerMainActivityComponent.builder().mainActivityMoudel(new MainActivityMoudel("可以")).build().inject(this);
    }

    @Override
    public void updateApkSucc(UpdateModel updateModel) {

    }
}

File: kotlin-frist-branch-TODO/app/src/main/java/com/mazouri/mvpkotlin/task/main/MainActivity.kt
package com.mazouri.mvpkotlin.task.main

import android.app.Activity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.view.View
import android.widget.Toast
import com.alibaba.android.arouter.facade.Postcard
import com.alibaba.android.arouter.facade.annotation.Autowired
import com.alibaba.android.arouter.facade.annotation.Route
import com.alibaba.android.arouter.facade.callback.NavigationCallback
import com.alibaba.android.arouter.launcher.ARouter
import com.mazouri.mvpkotlin.R
import com.mazouri.mvpkotlin.base.BaseActivity
import com.mazouri.mvpkotlin.base.ComponentHolder
import com.mazouri.mvpkotlin.base.Constants
import com.mazouri.mvpkotlin.data.model.Repository
import com.mazouri.mvpkotlin.utils.L
import kotlinx.android.synthetic.main.activity_main.*
import javax.inject.Inject

/**
 * Created by HeYingXin on 2018/4/16.
 */
@Route(path = Constants.Route.MAIN_ACTIVITY, group = Constants.Route.GROUP_ONE)
class MainActivity : BaseActivity(), MainContract.View {

    private var mAdapter: RepositoriesAdapter? = null

    @Inject lateinit var mMainPresenter: MainPresenter

    override val layout: Int
        get() = R.layout.activity_main

    @Autowired
    internal var key1: Long = 0

    @Autowired
    internal var key3: String? = null

    fun startMainActivity() {
        ARouter.getInstance().build(Constants.Route.MAIN_ACTIVITY, Constants.Route.GROUP_ONE).withLong("key1", 666L)
                .withString("key3", "888").navigation(ComponentHolder.getAppComponent().context(), object : NavigationCallback {
                    override fun onFound(postcard: Postcard) {
                        L.e("zhao", "onArrival: 找到了 ")
                    }

                    override fun onLost(postcard: Postcard) {
                        L.e("zhao", "onArrival: 找不到了 ")
                    }

                    override fun onArrival(postcard: Postcard) {
                        L.e("zhao", "onArrival: 跳转完了 ")
                    }

                    override fun onInterrupt(postcard: Postcard) {
                        L.e("zhao", "onArrival: 被拦截了 ")
                    }
                })
    }

    fun startMainActivity2(activity: Activity) {
        //直接通过所有拦截器并且制定requestCode
        ARouter.getInstance().build(Constants.Route.MAIN_ACTIVITY, Constants.Route.GROUP_ONE).withLong("key1", 666L)
                .withString("key3", "888").greenChannel().navigation(activity, Constants.IntentRequest.MAIN_CODE)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityComponent().inject(this)
        mMainPresenter.attachView(this)

        setupViews()

        showProgress()
        fab.setOnClickListener {
            showProgress()
            mMainPresenter.loadRepositories(getRepoUser())
        }
        mMainPresenter.loadRepositories(getRepoUser())
    }

    private fun getRepoUser(): String {
        return edit.text.toString().trim()
    }

    override fun onDestroy() {
        super.onDestroy()
        mMainPresenter.detachView()
    }

    private fun setupViews() {
        mAdapter = RepositoriesAdapter(ArrayList<Repository>(), {
            Toast.makeText(this, "您选择了" + it.name, Toast.LENGTH_SHORT).show()
        })

        recyclerViewRepositories.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        recyclerViewRepositories.adapter = mAdapter
    }

    private fun showProgress() {
        recyclerViewRepositories.visibility = View.GONE
        progress_bar.visibility = View.VISIBLE
        fab.isEnabled = false
    }

    private fun hideProgress() {
        recyclerViewRepositories.visibility = View.VISIBLE
        progress_bar.visibility = View.GONE
        fab.isEnabled = true
    }

    override fun showOrganizations(repositories: MutableList<Repository>) {
        mAdapter?.addRepositories(repositories)
        mAdapter?.notifyDataSetChanged()
        hideProgress()
    }

    override fun showLoadReposFailed(error: String?) {

        hideProgress()
    }

    override fun showError(error: String?) {

    }

    override fun showError(stringResId: Int) {

    }


}
